bundles/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/internal/socket/XiaomiBridgeSocket.java

```diff
@@ -42,7 +42,7 @@ public XiaomiBridgeSocket(int port) {
      */
     @Override
     synchronized DatagramSocket setupSocket() {
-        if (getOpenSockets().contains(getPort())) {
+        if (getOpenSockets().containsValue(getPort())) {
```

Comment:
I'm not sure this is correct.

Reply:
It should be! Because of the loose coupling `getOpenSockets()` now returns a `Map` instead of a `ConcurrentHashMap`. The `contains` method only exists on the `ConcurrentHashMap` but according to the JavaDocs it is considered legacy and calls `containsValue`:

```java
    // Hashtable legacy methods

    /**
     * Tests if some key maps into the specified value in this table.
     *
     * <p>Note that this method is identical in functionality to
     * {@link #containsValue(Object)}, and exists solely to ensure
     * full compatibility with class {@link java.util.Hashtable},
     * which supported this method prior to introduction of the
     * Java Collections Framework.
     *
     * @param  value a value to search for
     * @return {@code true} if and only if some key maps to the
     *         {@code value} argument in this table as
     *         determined by the {@code equals} method;
     *         {@code false} otherwise
     * @throws NullPointerException if the specified value is null
     */
    public boolean contains(Object value) {
        return containsValue(value);
    }
```